Adding awlneuss de

kueps opened the pull request:
Adding source for [AWL Neuss](https://buergerportal.awl-neuss.de/) in Neuss, Germany

- [x] Added `custom_components/waste_collection_schedule/waste_collection_schedule/source/awlneuss_de.py`
- [x] Added `doc/source/awlneuss_de.md`
- [x] Updated `README.md` via `update_docu_links.py`
- [x] Updated `info.md` via `update_docu_links.py`
- [x] Tested with `test_source.py`

```bash
Testing source awlneuss_de ...
  found 37 entries for Neuss, Theodor-Heuss-Platz 13
  found 37 entries for Neuss, Niederstrasse 42
```

5ila5 commented:
I'd prefer if the user can provide a street name instead of a street ID and fetching the ID from the JSON inside the script 

You could allow the use of street name or ID 

https://buergerportal.awl-neuss.de/api/v1/calendar/townarea-streets

kueps commented:
> I'd prefer if the user can provide a street name instead of a street ID and fetching the ID from the JSON inside the script
> 
> You could allow the use of street name or ID
> 
> https://buergerportal.awl-neuss.de/api/v1/calendar/townarea-streets

I added the option to provide a street name instead of the street ID. Now both is possible to provide the street ID or the street name. 
I added two test cases, see the result below:

```bash
Testing source awlneuss_de ...
  found 37 entries for Neuss, Theodor-Heuss-Platz 13
  found 37 entries for Neuss, Niederstrasse 42
  found 37 entries for Neuss, Bahnhofstrasse 67
  found 36 entries for Neuss, Bismarckstrasse 52
```

5ila5 commented:
Thanks for your contribution. 
Just did some minor reformatting (based on .pre-commit-config.yaml)

And fixed some type hints (str => list[str] for month_year and int => int | None for some others)
